geom/type_polygon.go

```diff
@@ -33,50 +22,85 @@ type Polygon struct {
 // any inner rings follow. If no rings are provided, then the returned Polygon
 // is the empty Polygon. The coordinate type of the polygon is the lowest
 // common coordinate type of its rings.
+//
+// An error is returned if any of the Polygon constraints are not met (see the
+// Validate method for details). Note that the validity of each input
+// LineString is not verified.
```

Comment:
I noticed validity of input geometries is not checked for certain geometries, esp. Multi* and here. Is this because validation is too computationally expensive to execute as default behaviour in a ctor?

Reply:
Sort of. We definitely don't want to validate them multiple times, because they're not cheap to run (especially the Polygon validation).

The idea is that Multi*, and GeometryCollection are all collection-style geometries, and provide additional constraints between elements of those collections. An argument could be made that Polygon should revalidate its input rings (which is cheap anyway) because it's _not_ a collection.

I'll have a bit more of a think about this.

Reply:
> I'll have a bit more of a think about this.

FYI I went with making the polygon constructor revalidate its rings (since polygons are categorically different than multi* types).